ovve_ui/utils/comms_link.py

```diff
@@ -0,0 +1,278 @@
+                
+                # DEBUG
+                print ('Received SEQ and CRC:')
+                print (in_pkt['sequence_count'])
+                print (in_pkt['crc'])
+                #END DEBUG
+                # UI may need this keeping
+                self.seq_num = in_pkt['sequence_count']
+                # Needed for the return packet
+                cmd_pkt['sequence_count'] = in_pkt['sequence_count']
+                # ENDIF
+
+            # ecu_settings_dict = {'to_JSON':ecu_settings_str, 'run_state': 0,'mode': 0,'tv': 0,'resp_rate':0 ,'ie_ratio': 0}
+            # self.comms_adapter.update_settings()
+
+            # create the return packet
+            endian = "little"
+            cmd_byteData = b""
+            #packet_version = 1
+
+            # get the updates from settings TODO: Make this event driven and only when callbackis called
+            cmd_pkt['mode_value'] = self.settings.mode
+            cmd_pkt['respiratory_rate_set'] = self.settings.resp_rate
+            cmd_pkt['tidal_volume_set'] = self.settings.tv
+            cmd_pkt['ie_ratio_set'] = self.settings.ie_ratio
+
+            cmd_byteData += bytes(cmd_pkt['sequence_count'].to_bytes(2, endian))
+            cmd_byteData += bytes(cmd_pkt['packet_version'].to_bytes(1, endian))
+            cmd_byteData += bytes(cmd_pkt['mode_value'].to_bytes(1, endian))
+            cmd_byteData += bytes(cmd_pkt['respiratory_rate_set'].to_bytes(4, endian))
+            cmd_byteData += bytes(cmd_pkt['tidal_volume_set'].to_bytes(4, endian))
+            cmd_byteData += bytes(cmd_pkt['ie_ratio_set'].to_bytes(4, endian))
+            
+            # TO DO set alarmbits correctly if sequence or CRC failed
+            cmd_byteData += bytes(cmd_pkt['alarm_bits'].to_bytes(4, endian))
+            calcSendCRC = self.crccitt(cmd_byteData.hex())
+            # DEBUG
+            # print ('CALC sent CRC HEX and int: ')
+            # print(calcSendCRC)
+            # print(int(calcSendCRC, 16))
+            #end DEBUG
+            cmd_pkt['crc']  = bytes.fromhex(calcSendCRC)
+            cmd_byteData += cmd_pkt['crc']
+            try:
+                self.ser.write(cmd_byteData)
+            except:
+                print("Faile to write to serial")
+            
+            # DEBUG
+            print ("Packet Written:")
+            print(''.join(r'\x'+hex(letter)[2:] for letter in cmd_byteData))
+            print("Sent back SEQ and CRC: ")
+            print (int.from_bytes(cmd_byteData[0:2], byteorder='little'))
+            print (int.from_bytes(cmd_byteData[20:], byteorder='big'))
+            # END DEBUG
+
+            #Update dict only if there is valid data
+            if validData == True:
+                # any settings set will not retunr correctly yet until Arduino sets set values correctly
+                # We can use the settings values like in simulator if so desired or maybe compare them
+                params_dict['run_state'] = self.settings.run_state
```

Comment:
We should be getting the actual run state from the MCU.  If UI sets run_state to "run" but Arduino doesn't actually start due to some problem, we don't want to show "running" in the UI

Reply:
We do not have a byte that tells me of the run state. I'll ask @mrrosen  and @CSTaylorJr

Reply:
I'll and this to open issue. I cannot implement this until I know what this looks like. Will not resolve in this PR

Reply:
https://github.com/OVVE/software-ui/issues/36